great_expectations/execution_engine/execution_engine.py

```diff
@@ -517,6 +517,77 @@ class MetricDomainTypes.
         """
         # Extracting value from enum if it is given for future computation
         domain_type = MetricDomainTypes(domain_type)
+
+        # Warning user if accessor keys are in any domain that is not of type table, will be ignored
+        if (
+            domain_type != MetricDomainTypes.TABLE
+            and accessor_keys is not None
+            and len(list(accessor_keys)) > 0
+        ):
+            logger.warning(
+                'Accessor keys ignored since Metric Domain Type is not "table"'
+            )
+
+        compute_domain_kwargs = copy.deepcopy(domain_kwargs)
+        accessor_domain_kwargs = {}
+        if domain_type == MetricDomainTypes.TABLE:
+            (
+                compute_domain_kwargs,
+                accessor_domain_kwargs,
+            ) = self._split_table_metric_domain_kwargs(
+                domain_kwargs=domain_kwargs,
+                domain_type=domain_type,
+                accessor_keys=accessor_keys,
+            )
+
+        elif domain_type == MetricDomainTypes.COLUMN:
+            (
+                compute_domain_kwargs,
+                accessor_domain_kwargs,
+            ) = self._split_column_metric_domain_kwargs(
```

Comment:
with this level of verbosity, I don't think we've saved ourselves any effort by using a Tuple. I'd consider replacing the return type with a simple class that is more descriptive; the class name could be "SplitDomainKwargs" with attributes for "compute" and "accessor"